locations/spiders/pick_n_pay.py

```diff
@@ -0,0 +1,49 @@
+import scrapy
+
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, OpeningHours
+
+
+class PickNPaySpider(scrapy.Spider):
+    name = "pick_n_pay"
+    item_attributes = {"brand": "Pick n Pay", "brand_wikidata": "Q7190735"}
+    start_urls = ["https://api.pnp.co.za/stores/v2"]
+
+    def parse(self, response, **kwargs):
+        for store in response.json():
+            store["id"] = store.pop("storeId")
+            address = store["storeAddress"]
+            store["housenumber"] = address.get("streetNo")
+            store["street"] = address.get("street")
+            store["suburb"] = address.get("suburb")
+            store["city"] = address.get("city")
+            province = address.get("province") or {}  # can be JSON null
+            store["province"] = province.get("name")
+            store["country"] = address.get("country")
+            store["postcode"] = address.get("postalCode")
```

Comment:
How many of these lines are necessary in front of `DictParser.parse`? Shouldn't DictParser be able to detect fields such as `storeId` being a `ref` field value, etc? Can this list be reduced so that DictParser does the heavy lifting?

Reply:
Thanks, I managed to make it _much_ simpler now :)